hibernate-core/src/main/java/org/hibernate/internal/CoreMessageLogger.java

```diff
@@ -1848,4 +1848,8 @@ void attemptToAssociateProxyWithTwoOpenSessions(
 	@LogMessage(level = WARN)
 	@Message(value = "Using @AttributeOverride or @AttributeOverrides in conjunction with entity inheritance is not supported: %s. The overriding definitions are ignored.", id = 499)
 	void unsupportedAttributeOverrideWithEntityInheritance(String entityName);
+
+	@LogMessage(level = WARN)
+	@Message(value = "The [%s] property of the [%s] entity was modified, but it won't be updated because the property is immutable!", id = 500)
```

Comment:
This is not a proper review but I think we should remove the exclamation point. It does not seem very professional to me.

Another thing is that we should check the full output of the Hibernate tests to check that this error message is not logged when it should not be.